os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -177,13 +197,24 @@ int mm_check_heap_corruption(struct mm_heap_s *heap)
 					dbg("Corrupted node flink(0x%08x) and next node blink(0x%08x) do not match\n", ((struct mm_freenode_s *)node)->flink,
 							((struct mm_freenode_s *)node)->flink->blink);
 					dbg("#########################################################################################\n");
-					mm_givesemaphore(heap);
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
```

Comment:
Could you let know why user context does not require this?

Reply:
The API up_interruptcontext is not present in user side. 
Also, if we are currently executing in user side, then it is impossible to be in irq mode.